WordPress/wordpress-develop
# Try making server aware of block variants

https://core.trac.wordpress.org/ticket/52688

Experiment, required for https://github.com/WordPress/gutenberg/pull/29095

This PR exposes `variations` in the block type definition and adds two new labels `item_link` and `item_link_description` for post types and taxonomies.

Changes here will allow us to populate link navigation variations via `register_block_type_from_metadata` without needing to update the post type and taxonomy REST APIs to allow filtering by `show_in_nav_menus`.

### Testing Instructions
- Checkout and run this branch locally with an environment of your choice. (Don't forget to `npm run build`)
- Test the GET REST endpoint `/wp/v2/block-types`. Notice how there is a new variations field returned.
- Make sure local tests pass by running `npm run test:php -- --filter REST_Block_Type_Controller_Test`

<img width="951" alt="Screen Shot 2021-02-24 at 8 31 02 PM" src="https://user-images.githubusercontent.com/1270189/109103202-49a0ed80-76df-11eb-8238-6e6fb389411a.png">

If you like, also follow test instructions in https://github.com/WordPress/gutenberg/pull/29095.

https://user-images.githubusercontent.com/1270189/109558000-b32d4d00-7a8d-11eb-8c23-64fb90ed9c6e.mp4

### Default label behavior

<img width="1397" alt="Screen Shot 2021-03-01 at 12 34 45 PM" src="https://user-images.githubusercontent.com/1270189/109559265-4d41c500-7a8f-11eb-8e5d-6dca934d8188.png">
